### PR TITLE
[KOGITO-712] Kogito rule getting started tutorial

### PR DESCRIPTION
**Pull request by tkobayas**
This is the first draft of get-started examples for KOGITO-712. I'm open to any modification.

### CODE REVIEW

**Comment by tkobayas** [comment]
This example is associated to documents in https://github.com/kiegroup/kogito-website/pull/22

File: ruleunit-quarkus-example/src/main/resources/META-INF/kmodule.xml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<kmodule xmlns="http://www.drools.org/xsd/kmodule"/>
```
Comment (evacchi) [inline ruleunit-quarkus-example/src/main/resources/META-INF/kmodule.xml]: you can delete this file, it's not required in Kogito

File: ruleunit-springboot-example/src/main/resources/META-INF/kmodule.xml [hunk 2]
```diff
@@ -0,0 +1,2 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<kmodule xmlns="http://www.drools.org/xsd/kmodule"/>
```
Comment (evacchi) [inline ruleunit-springboot-example/src/main/resources/META-INF/kmodule.xml]: same here, you can delete this file, it's not required in Kogito

File: ruleunit-springboot-example/src/main/resources/META-INF/resources/index.html [hunk 3]
```diff
@@ -0,0 +1,152 @@
+<!DOCTYPE html>
```
Comment (evacchi) [inline ruleunit-springboot-example/src/main/resources/META-INF/resources/index.html]: you may want to delete or change this file, we don't want to say Welcome to Quarkus in Spring Boot :-)

**Comment by tkobayas** [comment]
Removed kmodule.xml from both examples. Removed index.html from ruleunit-springboot-example. Thank you for the comments, @evacchi

File: ruleunit-quarkus-example/src/main/java/org/kie/kogito/queries/LoanUnit.java [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+package org.kie.kogito.queries;
+
+import java.io.Serializable;
+
+import org.kie.kogito.conf.SessionsPool;
+import org.kie.kogito.rules.DataSource;
+import org.kie.kogito.rules.DataStore;
+import org.kie.kogito.rules.RuleUnitData;
+
+@SessionsPool(1)
```
Comment (mariofusco) [inline ruleunit-quarkus-example/src/main/java/org/kie/kogito/queries/LoanUnit.java]: Please remove this annotation. If we want to promote this as main example for rule unit this annotation would be at best distracting.

File: ruleunit-quarkus-example/src/main/java/org/kie/kogito/queries/LoanUnit.java [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+public class LoanUnit implements RuleUnitData, Serializable {
```
Comment (mariofusco) [inline ruleunit-quarkus-example/src/main/java/org/kie/kogito/queries/LoanUnit.java]: Implementing Serializable is not necessary here, please remove it. Same thing for the serialVersionUID.

**Comment by mariofusco** [comment]
@tkobayas The example is ok. Please just remove the minor things that I mentioned in my other comments and if possible also add a test exercising directly the rest endpoint. For the quarkus based implementation you could do something similar to this https://github.com/kiegroup/kogito-examples/blob/master/dmn-quarkus-example/src/test/java/org/kie/dmn/kogito/quarkus/example/TrafficViolationTest.java I guess it's possible to do something similar also on the spring-boot side, but I don't know the details.

File: ruleunit-quarkus-example/src/main/java/org/kie/kogito/queries/LoanUnit.java (hunk 4 above)
Comment (tkobayas) [reply ruleunit-quarkus-example/src/main/java/org/kie/kogito/queries/LoanUnit.java]: Fixed. Thanks!

File: ruleunit-quarkus-example/src/main/java/org/kie/kogito/queries/LoanUnit.java (hunk 5 above)
Comment (tkobayas) [reply ruleunit-quarkus-example/src/main/java/org/kie/kogito/queries/LoanUnit.java]: Fixed. Thanks!

**Comment by tkobayas** [comment]
@mariofusco Thank you for the suggestions. I applied the fixes and added Rest test cases for both quarkus and spring boot.

**Review by evacchi — APPROVED** [review]
Mario and I took a look at this together, so if you address his comments we are good to go!

File: ruleunit-quarkus-example/src/main/resources/META-INF/kmodule.xml (hunk 1 above)
Comment (tkobayas) [reply ruleunit-quarkus-example/src/main/resources/META-INF/kmodule.xml]: Deleted, thx!

File: ruleunit-springboot-example/src/main/resources/META-INF/kmodule.xml (hunk 2 above)
Comment (tkobayas) [reply ruleunit-springboot-example/src/main/resources/META-INF/kmodule.xml]: Deleted, thx!

File: ruleunit-springboot-example/src/main/resources/META-INF/resources/index.html (hunk 3 above)
Comment (tkobayas) [reply ruleunit-springboot-example/src/main/resources/META-INF/resources/index.html]: Deleted. Also deleted in quarkus as it's not important. Thanks!

**Comment by tkobayas** [comment]
Applied suggestions from Mario and Edoardo.

@baldimir Do you have any review comment especially about unit tests?
QueryTest : Test using generated endpoint classes
RestQueryTest: Test using RestAssured

**Comment by tkobayas** [comment]
Hi @MarianMacik , Could you review this especially about unit tests? Thanks!
QueryTest : Test using generated endpoint classes
RestQueryTest: Test using RestAssured

**Review by MarianMacik — APPROVED** [review]
Everything looks good to me. Thank you!

**Comment by tkobayas** [comment]
Thanks all! Could you merge this? @mariofusco

**Pull request merged**